app/scripts/controllers/user-storage/encryption.ts

```diff
@@ -23,14 +36,57 @@ function bytesToUtf8(byteArray: Uint8Array) {
   return decoder.decode(byteArray);
 }
 
+/**
+ * In Memory Cache of the derived key from a given salt.
+ * We can tidy this to make it reusable with future versions and KDF functions
+ */
+const inMemCachedKDF: Map<string, Uint8Array> = new Map();
+
+type CachedEntry = {
+  salt: Uint8Array;
+  key: Uint8Array;
+};
+const getCachedKeyBySalt = (salt: Uint8Array): CachedEntry | undefined => {
+  const base64Salt = byteArrayToBase64(salt);
+  const cachedKey = inMemCachedKDF.get(base64Salt);
+  if (!cachedKey) {
+    return undefined;
+  }
+
+  return {
+    salt,
+    key: cachedKey,
+  };
+};
+const getAnyCachedKey = (): CachedEntry | undefined => {
+  const cachedEntry = Array.from(inMemCachedKDF.entries()).at(0);
+  if (!cachedEntry) {
+    return undefined;
+  }
+
+  const bytesSalt = base64ToByteArray(cachedEntry[0]);
+  return {
+    salt: bytesSalt,
+    key: cachedEntry[1],
+  };
+};
+const setCachedKey = (salt: Uint8Array, key: Uint8Array): void => {
+  const base64Salt = byteArrayToBase64(salt);
+  inMemCachedKDF.set(base64Salt, key);
+};
+
 class EncryptorDecryptor {
   #ALGORITHM_NONCE_SIZE: number = 12; // 12 bytes
 
   #ALGORITHM_KEY_SIZE: number = 16; // 16 bytes
 
-  #PBKDF2_SALT_SIZE: number = 16; // 16 bytes
+  #SCRYPT_SALT_SIZE: number = 16; // 16 bytes
+
+  #SCRYPT_N: number = 2 ** 17; // CPU/memory cost parameter (must be a power of 2, > 1)
+
+  #SCRYPT_r: number = 8; // Block size parameter
 
-  #PBKDF2_ITERATIONS: number = 900_000;
+  #SCRYPT_p: number = 1; // Parallelization parameter
```

Comment:
Using [OWASP configuration](https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#scrypt)

This is flexible if the "work" lever needs to be increased.